src/boundary_conditions.jl

```diff
@@ -94,36 +156,40 @@ getbc(cbc::CBC, ::Val{S}) where S = getfield(cbc, S)
 getbc(cbc::CBC, ::Val{:bottom}) = getfield(cbc, :right)
 getbc(cbc::CBC, ::Val{:top}) = getfield(cbc, :left)
 
-
 #####
 ##### Boundary conditions for Fields
 #####
 
 """
-    FieldBoundaryConditions(x, y, z)
+    FieldBoundaryConditions
 
-Construct `FieldBoundaryConditions` for a field.
-Field boundary conditions have `CoordinateBoundaryCondition`s in
-`x`, `y`, and `z`.
+A type describing a set of three `CoordinateBoundaryCondition`s stored in a named tuple and
+designed to be applied to a field along each of the three coordinate `:x`, `:y`, and `:z`.
 """
 const FieldBoundaryConditions = NamedTuple{(:x, :y, :z)}
 
-FieldBoundaryConditions(x, y, z) = FieldBoundaryConditions((x, y, z))
+"""
+    FieldBoundaryConditions(x, y, z)
 
-function FieldBoundaryConditions(;
-    x = CoordinateBoundaryConditions(),
-    y = CoordinateBoundaryConditions(),
-    z = CoordinateBoundaryConditions())
-    return FieldBoundaryConditions(x, y, z)
-end
+Construct a `FieldBoundaryConditions` using a `CoordinateBoundaryCondition` for each of the
+`x`, `y`, and `z` coordinates.
+"""
+FieldBoundaryConditions(x, y, z) = FieldBoundaryConditions((x, y, z))
```

Comment:
I believe this pattern will show up in the docs something like

```julia
julia> ?FieldBoundaryConditions

    FieldBoundaryConditions

A type describing a set of three `CoordinateBoundaryCondition`s stored in a named tuple and
designed to be applied to a field along each of the three coordinate `:x`, `:y`, and `:z`.

    FieldBoundaryConditions(x, y, z)

Construct a `FieldBoundaryConditions` using a `CoordinateBoundaryCondition` for each of the 
`x`, `y`, and `z` coordinates.
```

Is this what we want users to see? Or would it be better for them to see

```julia
julia> ?FieldBoundaryConditions

    FieldBoundaryConditions(x, y, z)

Construct a `FieldBoundaryConditions` using a `CoordinateBoundaryCondition` for each of the 
`x`, `y`, and `z` coordinates.
```

I feel the first is more information than a user needs, so the second could be preferred for conciseness. But as with many questions of documentation, this is largely a matter of opinion.